### ISSUE TITLE
Proj0044 triggered even when `ContinuousIntegrationBuild` is defined with a condition in `Directory.Build.props` to only be included in CI

### ISSUE THREAD

**Issue by Frulfump** [dotnet-project-file-analyzers/dotnet-project-file-analyzers#756]
I have this in a `Directory.Build.props` in my root   

```xml
...
<ContinuousIntegrationBuild Condition="'$(GITHUB_ACTIONS)' == 'true'">true</ContinuousIntegrationBuild>
<RestorePackagesWithLockFile>true</RestorePackagesWithLockFile>
...
```

But I still get Proj0044 reported for a project file when doing a local build (not in GitHub Actions)
https://dotnet-project-file-analyzers.github.io/rules/Proj0044.html

The condition means the property isn't included when it's not true

**Comment by Corniel** [comment]
You miss the `<RestoreLockedMode>` property?!

**Comment by Frulfump** [comment]
> You miss the `<RestoreLockedMode>` property?!

I run with locked mode in CI with the command line option. I think this FP is triggered due to the condition not being taken into account?
UPDATE: I have this in my `Directory.Build.props` as well `<RestoreLockedMode Condition="'$(GITHUB_ACTIONS)' == 'true'">true</RestoreLockedMode>`

**Comment by Corniel** [comment]
With `--locked-mode` you mean? I never tested if a rule could check that, I assume it does, and that the value is known in MSBuild.

**Comment by Corniel** [comment]
@Frulfump can you review the changes?

**Comment by Frulfump** [comment]
> With `--locked-mode` you mean? I never tested if a rule could check that, I assume it does, and that the value is known in MSBuild.

Yes exactly!

**Comment by Frulfump** [comment]
This seems to still happen with 1.12.0
>warning Proj0044: Define the <RestoreLockedMode> node with value 'true' when <ContinuousIntegrationBuild> is enabled (https://dotnet-project-file-analyzers.github.io/rules/Proj0044.html)

` <ContinuousIntegrationBuild>` isn't included as there's a condition for it to only be included and set to `true` when I'm running in GitHub Actions but I still see it, so maybe the detection code is ignoring that the property is conditioned?

**Comment by Corniel** [comment]
I'll look into this.

**Comment by Frulfump** [comment]
> I'll look into this.

Thanks! I updated the title to better reflect the actual issue.
EDIT: Related to https://github.com/dotnet-project-file-analyzers/dotnet-project-file-analyzers/issues/33 but the issue here seems to be that the condition is ignored and the analyzer thinks the value is set to `true` even though the property will actually not be included at all so maybe possible to also inspect the resolved value (or only look at the resolved value(the property might not be set at all?))

**Comment by Corniel** [comment]
@Frulfump Does this issue still occur?

**Comment by Frulfump** [comment]
@Corniel nope, and I can't get it to repro on 1.12.0 either. Closing as completed and will re-open if I see it again
